bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/IFindReplaceLogic.java

```diff
@@ -25,6 +25,20 @@
  */
 public interface IFindReplaceLogic {
 
+	/**
+	 * Sets the string to be used for searching in find and replace operations.
+	 *
+	 * @param findString the find string to use, must not be null
+	 */
+	public void setFindString(String findString);
```

Comment:
Somehow this method does more than I would expect just from reading this header.
The method
1) performs a null-check
2) updates incremental search
3) thus invalidates the search status

I am not sure if the name "setFindString" is good or not, however the javadocs should definitely somehow reflect these side-effects

Reply:
That you for the catch! Indeed, the contract of `setFindString()` needs to be updated accordingly.

The null check is conforming to the method's documentation, requiring the parameter not to be null. The null check only ensures that it fails as early as possible.

The update of the incremental search and the change of the status was not documented yet. These are no "side effects" but intended effects when setting the find string while incremental search is active, so in my opinion the method's implementation and name are fine and only the contract needs to be updated accordingly. I have done that in 9978946.

Reply:
The new comment looks good now, it clarifies well what happens under the hood.